### PR TITLE
Feature/v2

### PR DESCRIPTION
**Pull request by drobnikj**
Sorry for too long PR

### CODE REVIEW

File: src/apify_helpers.js [hunk 1]
```diff
@@ -37,25 +46,31 @@ const getValuesFromKeyValueStore = async (z, storeId, keys) => {
...
             .request(`${APIFY_API_ENDPOINTS.keyValueStores}/${storeId}/records/${key}`)
             .then((response) => {
+                if (response.status === 404) {
+                    values[key] = {
+                        error: `Cannot find ${key} in key-value store`,
+                    };
+                    return;
+                }
                 try {
                     const maybeObject = JSON.parse(response.content);
```
Comment (mtrunkat) [inline src/apify_helpers.js]: Shouldn't we support also text types or possibly other types (images, ...)?

File: src/apify_helpers.js [hunk 2]
```diff
@@ -64,11 +79,172 @@ const enrichTaskRun = async (z, run, storeKeysToInclude = []) => {
         run = Object.assign({}, run, keyValueStoreValues);
     }
 
-    if (defaultDatasetId) run.datasetItems = await getDatasetItems(z, defaultDatasetId, 500);
+    if (defaultDatasetId) run.datasetItems = await getDatasetItems(z, defaultDatasetId, { limit: 500 }, run.actId);
```
Comment (mtrunkat) [inline src/apify_helpers.js]: `500` should be constant somwhere

File: src/apify_helpers.js [hunk 3]
```diff
@@ -64,11 +79,172 @@ const enrichTaskRun = async (z, run, storeKeysToInclude = []) => {
...
+        method: 'DELETE',
+    });
+
+    return {};
+};
+
+// Gets actor run from bundle clean request and enriches it.
+const getActorRun = async (z, bundle) => {
+    const run = bundle.cleanedRequest.resource;
+    const enrichRun = await enrichActorRun(z, run);
```
Comment (mtrunkat) [inline src/apify_helpers.js]: enrich**ed**Run

File: src/apify_helpers.js [hunk 4]
```diff
@@ -64,11 +79,172 @@ const enrichTaskRun = async (z, run, storeKeysToInclude = []) => {
...
+const getActorAdditionalFields = async (z, bundle) => {
+    const { actorId } = bundle.inputData;
+    if (!actorId) return [];
+
+    const actorResponse = await wrapRequestWithRetries(z.request, {
+        url: `${APIFY_API_ENDPOINTS.actors}/${actorId}`,
+    });
+
+    /*
+    TODO: We need to fetch input schema and prefill input regarding that. But we need to have
```
Comment (mtrunkat) [inline src/apify_helpers.js]: Let's create issues here at Github instead of TODOs

File: src/apify_helpers.js [hunk 5]
```diff
@@ -64,11 +79,172 @@ const enrichTaskRun = async (z, run, storeKeysToInclude = []) => {
...
+    input schema in build detail API. Now we use just defaultRunOptions, exampleRunInput.
+     */
+
+    const actor = actorResponse.json;
+    const { build, timeoutSecs, memoryMbytes } = actor.defaultRunOptions;
+    const defaultActorBuildTag = build || BUILD_TAG_LATEST;
+
+    // Parse and stringify json input body if there is
+    let inputBody; let inputContentType;
```
Comment (mtrunkat) [inline src/apify_helpers.js]: Declarations should be on separate lines.

File: src/consts.js [hunk 6]
```diff
@@ -47,11 +49,32 @@ const TASK_OUTPUT_FIELDS = [
     { key: 'defaultRequestQueueId', label: 'Default request queue ID', type: 'string' },
 ];
 
+const TASK_RUN_SAMPLE = Object.assign({ actorTaskId: 'UJNG9zau8PEB7U' }, ACTOR_RUN_SAMPLE);
+
+const TASK_RUN_OUTPUT_FIELDS = ACTOR_RUN_OUTPUT_FIELDS.concat([{ key: 'actorTaskId', label: 'Actor task ID', type: 'string' }]);
+
 const DEFAULT_KEY_VALUE_STORE_KEYS = ['OUTPUT'];
```
Comment (mtrunkat) [inline src/consts.js]: `OUTPUT` constant is in apify shared

File: src/consts.js [hunk 7]
```diff
@@ -47,11 +49,32 @@ const TASK_OUTPUT_FIELDS = [
...
 
+const DEFAULT_PAGINATION_LIMIT = 100;
+
+// Actor ID of apify/legacy-phantomjs-crawler
+const LEGACY_PHANTOM_JS_CRAWLER_ID = 'YPh5JENjSSR6vBf2E';
```
Comment (mtrunkat) [inline src/consts.js]: We should move this to shared also. Ill try to keep it in mind when I do some consts changes.

File: src/creates/actor_run.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            'Content-Type': inputContentType,
+        };
+    }
+    if (inputBody) requestOpts.body = inputBody;
+
+    const runResponse = await wrapRequestWithRetries(z.request, requestOpts);
+
+    let run = runResponse.json;
+    if (runSync) {
+        run = await enrichActorRun(z, run);
```
Comment (mtrunkat) [inline src/creates/actor_run.js]: You can't be sure that run finished at this point. Should we throw some error? Or wait another 120s? what is the limit?

File: src/creates/set_value.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+const { getOrCreateKeyValueStore } = require('../apify_helpers');
+
+const setValue = async (z, bundle) => {
+    const { storeIdOrName, key, value } = bundle.inputData;
+    const store = await getOrCreateKeyValueStore(z, storeIdOrName);
+    const keyValueStoreValueUrl = `${APIFY_API_ENDPOINTS.keyValueStores}/${store.id}/records/${key}`;
+
+    let valueObject;
+    try {
+        valueObject = JSON.parse(value);
```
Comment (mtrunkat) [inline src/creates/set_value.js]: You can throw even before if value is not a string.

File: src/searches/actor_last_run.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+            url: `${APIFY_API_ENDPOINTS.actors}/${actorId}/runs/last`,
+            params: status ? { status } : {},
+        });
+    } catch (err) {
+        if (err.message.includes('not found')) return [];
+
+        throw err;
+    }
+
+    if (!lastActorRunResponse.json) return [];
```
Comment (mtrunkat) [inline src/searches/actor_last_run.js]: Can this happen?

File: src/searches/fetch_items.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            url: `${APIFY_API_ENDPOINTS.datasets}`,
+            method: 'POST',
+            params: {
+                name: datasetIdOrName,
+            },
+        });
+        dataset = storeResponse.json;
+    }
+
+    // NOTE: Because testing user had _id instead of id in data and we run integration tests under this user.
```
Comment (mtrunkat) [inline src/searches/fetch_items.js]: Can't we change this?

File: src/apify_helpers.js (hunk 1 above)
Comment (drobnikj) [reply src/apify_helpers.js]: I think this will be for next version, I created issue of it. #8

File: src/apify_helpers.js (hunk 4 above)
Comment (drobnikj) [reply src/apify_helpers.js]: #9

File: src/creates/set_value.js (hunk 9 above)
Comment (drobnikj) [reply src/creates/set_value.js]: It is always a string, it was validated on Zapier UI as I set up to the field to be `type: 'text'`

File: src/creates/actor_run.js (hunk 8 above)
Comment (drobnikj) [reply src/creates/actor_run.js]: I have to test it in Zapier which error we can get on UI. I will update this in next version.

**Pull request merged**